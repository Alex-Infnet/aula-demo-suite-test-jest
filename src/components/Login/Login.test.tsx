import renderer from 'react-test-renderer';
import {describe, expect, it} from '@jest/globals'
import { Login } from "./index";

describe("Snapshot tests for login page", () => {
    it("Should show the form only", () => {

        const component = renderer.create(
            <Login />
        )

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    })

    it("Should show a message when the user does not provide username", () => {

        const component = renderer.create(
            <Login />
        )       

        renderer.act(() => {
            component.root.findByType('button').props.onClick()
        })

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    })
})