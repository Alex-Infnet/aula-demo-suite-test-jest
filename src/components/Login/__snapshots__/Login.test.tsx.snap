// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for login page Should show a message when the user does not provide username 1`] = `
<form>
  <div>
    <p>
      Nome de usuário deve ser informado
    </p>
  </div>
  <div>
    <label
      htmlFor="username"
    >
      Usuário
    </label>
    <input
      id="username"
      name="username"
      placeholder="Please, inform the username"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="password"
    >
      Senha
    </label>
    <input
      id="password"
      name="password"
      placeholder="******"
      type="password"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Entrar
  </button>
</form>
`;

exports[`Snapshot tests for login page Should show the form only 1`] = `
<form>
  <div>
    <label
      htmlFor="username"
    >
      Usuário
    </label>
    <input
      id="username"
      name="username"
      placeholder="Please, inform the username"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="password"
    >
      Senha
    </label>
    <input
      id="password"
      name="password"
      placeholder="******"
      type="password"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Entrar
  </button>
</form>
`;
